dvc/cli.py

```diff
@@ -204,6 +278,21 @@ def get_main_parser():
     for cmd in COMMANDS:
         cmd.add_parser(subparsers, parent_parser)
 
+    for cmd, subparser in subparsers.choices.items():
+        if cmd in parser.hidden_cmds:
+            continue
+        parser.cmd_choices[cmd] = []
```

Comment:
It is a little weird that we parse it outside of parser itself and then set the `cmd_choices` here. Feels like it should be done in the parser itself, because it looks to be spreading the context of this feature without a good reason.

Reply:
Are you suggesting moving this logic into a method on the `DvcParser` class? The reason this was done outside of the parser was because we have access to `subparsers` here but maybe we can pass that in as a parameter.

Reply:
Hm, aren't those also part of the root parser?

Reply:
These were part of the root parser 👍 updated this.